api_app/admin.py

```diff
@@ -28,36 +28,46 @@ class TagAdminView(GuardedModelAdmin):
 
 
 # Auth Token stuff
-class CustomTokenAdmin(TokenAdmin):
+class CustomAuthTokenAdmin(AuthTokenAdmin):
     """
-    Custom admin view for TokenAdmin model
+    Custom admin view for AuthToken model
     """
 
-    # searchable fields
-    search_fields = ("user__username",)
-
+    fieldsets = []
+    exclude = []
+    raw_id_fields = ("user",)
+    readonly_fields = ("token", "expiry", "client")
     __fieldsets_custom = [
         (
-            "Create API Token For",
+            "Create token for PyIntelOwl",
             {
                 "fields": ("user",),
                 "description": """
                     <h3>Token will be auto-generated on save.</h3>
-                    <h5>You can use this auth token with the PyIntelOwl client
-                     or normal HTTP requests too.</h5>
+                    <h3>This token will be valid for 10 years.</h3>
                 """,
             },
         ),
     ]
 
     def add_view(self, request, extra_content=None):
         self.fieldsets = self.__fieldsets_custom
-        return super(CustomTokenAdmin, self).add_view(request)
+        return super(CustomAuthTokenAdmin, self).add_view(request)
+
+    def has_change_permission(self, *args, **kwargs):
+        return False
+
+    def save_model(self, request, obj, form, change):
+        client = Client.objects.get(name="pyintelowl")
+        # lgtm [py/unused-local-variable]
+        obj = AuthToken.objects.create(user=obj.user, client=client)
```

Comment:
LGTM points that this variable is unused.

Reply:
it's used by django admin internally, that's why I put the `# lgtm...` ignore flag